### PR TITLE
feat(pip_parse)!: use user friendly aliases to pip repos in requirements arrays

### PR DESCRIPTION
**Pull request by aignas**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently the labels in the `all_requirements` and `all_whl_requirements` are not useable, which is needed for gazelle bzlmod support.

Issue number: #965

## What is the new behavior?

We change the labels to point to the aliases that are present when `bzlmod` is used.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: examples/bzlmod/BUILD.bazel [hunk 1]
```diff
@@ -14,7 +14,8 @@ py_library(
     name = "lib",
     srcs = ["__init__.py"],
     deps = [
-        requirement("pylint"),
+        # You can reference libraries using plain text form or the 'requirement' macro
+        "@pip//:pylint_pkg",
```
Comment (alexeagle) [inline examples/bzlmod/BUILD.bazel:18]: can it just be `@pip//pylint` ? That's the nicest syntax users would enjoy most.
 (we'd generate a `external/pip/pylint/BUILD.bazel` file with appropriate aliases)

**Review by alexeagle — COMMENTED** [review]
What if gazelle was changed to use the `requirement` function?

That's how it works in Java:
https://github.com/bazelbuild/rules_jvm_external/blob/master/examples/bzlmod/java/src/com/github/rules_jvm_external/examples/bzlmod/BUILD#L7

**Comment by groodt** [comment]
> What if gazelle was changed to use the requirement function?
> 
> That's how it works in Java:
https://github.com/bazelbuild/rules_jvm_external/blob/master/examples/bzlmod/java/src/com/github/rules_jvm_external/examples/bzlmod/BUILD#L7

I am working on a PR that extracts the external package artifact macros out of the repository. Which would make things a lot simpler for both bzlmod and general maintenance of the rules. The artifact macros in rules_jvm_external work quite differently to the present repository rules in rules_python.

File: examples/bzlmod/BUILD.bazel (hunk 1 above)
Comment (aignas) [reply examples/bzlmod/BUILD.bazel:18]: I would love to use this and would be willing to implement this if this approach was agreed universally. @rickeylev, any opinions here?

**Comment by aignas** [comment]
@groodt, should we wait for the #989 to be done before we think about this again?

I personally found that something like `@pip//pylint` would be much nicer than `artifact("pylint")` because I can easily query the external targets as there are fewer layers of abstraction here. Any opinions from others?

File: examples/bzlmod/BUILD.bazel (hunk 1 above)
Comment (aignas) [reply examples/bzlmod/BUILD.bazel]: @alexeagle, what do you think about the inclusion of the user friendly aliases?

**Comment by aignas** [comment]
This is getting stale, so I'll close this in favour of something else in the somewhat near future.

**Pull request closed without merging**